transformer_engine/jax/csrc/modules.h

```diff
@@ -56,28 +63,42 @@ struct CustomCallGemmDescriptor {
     size_t m;
     size_t n;
     size_t k;
+    size_t workspace;
     DType A_dtype;
     DType B_dtype;
     DType D_dtype;
+    DType work_dtype;
     bool transa;
     bool transb;
     bool use_split_accumulator;
 };
 
-pybind11::bytes PackCustomCallGemmDescriptor(size_t m, size_t n, size_t k, DType A_dtype,
-                                             DType B_dtype, DType D_dtype, bool transa, bool transb,
+pybind11::bytes PackCustomCallGemmDescriptor(size_t m, size_t n, size_t k, size_t workspace,
+                                             DType A_dtype, DType B_dtype, DType D_dtype,
+                                             DType work_dtype, bool transa, bool transb,
                                              bool use_split_accumulator);
 
 struct CustomCallNormDescriptor {
     size_t n;
     size_t hidden;
+    size_t workspace;
+    size_t barrier;
+    size_t dgamma_part;
+    size_t *dbeta_part;
```

Comment:
Could you explain why only `dbeta_part` is a pointer?

Reply:
The shape for this partial is 2D so I store the sizing as an array here. We could just store it as two separate size_t variables if there's a reason why an array here is undesirable. But otherwise using an array keeps the Python interface consistent with the shape of other workspace tensors.

Reply:
Note that dgamma_part also has a 2D shape

Reply:
Okay, that's going to need to be a pointer too then. I didn't notice it was 2D. Thanks for the heads up!